internal/provider/utils.go

```diff
@@ -15,8 +16,23 @@ import (
 	"github.com/hashicorp/terraform-plugin-framework/types"
 )
 
+func ReturnAAPNamedURL(id types.Int64, name types.String, orgName types.String, URI string) (string, error) {
+	if IsValueProvided(id) {
```

Comment:
because each of these cases is immediately returning, there is no need for the `if, else if` blocks 

Also, is there ever a case where we want all 3, `id, name, orgName` in the URI?